source/extensions/filters/listener/original_dst/config.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "envoy/extensions/filters/listener/original_dst/v3/original_dst.pb.h"
+#include "envoy/extensions/filters/listener/original_dst/v3/original_dst.pb.validate.h"
+#include "envoy/network/filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace OriginalDst {
+
+class Config {
+public:
+  Config() = default;
+  Config(const envoy::extensions::filters::listener::original_dst::v3::OriginalDst& config,
+         const envoy::config::core::v3::TrafficDirection& traffic_direction)
+      : method_(getOriginalDstMethod(config)), traffic_direction_(traffic_direction) {}
+
+  enum class OriginalDstMethod {
+    SocketOption = 0,
+    NoOp = 1,
```

Comment:
Please change `NoOp` to `CurrentLocalAddress` for consistency.  It would be also good to document the meaning of these values.

The disparity between `DNAT` in the proto definition vs `SocketOption` in this enum is also odd.  It would be good to reconcile naming.